CMSgov/mint-app
# [EASI-3810] - FE CR TDL

# EASI-3810

[Figma](https://www.figma.com/file/M72DReE73SSXMN48WZfaKN/MINT---New-Model-Process-(Model-Intake)?node-id=6891%3A60341&mode=dev)

## Changes and Description

- Updated queries for fetching all crtdls
  - Now split into `crs` and `tdls`
- Added new mutations for creating/getting/updating/deleting CR's and TDL's
- Converted all CRTDL queries into graphql-codegen
- Update CRTDL Table to split into 2 tables
- Updated form/validation for selecting either CR or TDL to add/update
- Text changes to form/table/readonly views
- Updated unit tests
- Updated cypress tests



## How to test this change

- Clean db
- Create a plan
- Navigate to task list and slect to add a CR or TDL
- Verify text changes on table page
- Click to add a CR or TDL
- Fill out form for one/each type
- Verify validation and dat persistence
- Verify table data is accurate
- Verify pagination after 5 rows
- Verify readonly
- Verify favorite cards render correctly

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
